app/models/user.rb

```diff
@@ -632,43 +570,9 @@ def send_devise_notification(notification, *args)
 
   private
 
-  def sync_relevant_profile_fields_to_user_settings_table(users_setting_record)
-    PROFILE_FIELDS_TO_MIGRATE_TO_USERS_SETTINGS_TABLE.each do |field|
-      # rubocop:disable Layout/LineLength
-      users_setting_record.assign_attributes(field => profile.public_send(field)) if profile&.public_send(field).present?
-      # rubocop:enable Layout/LineLength
-    end
-  end
-
-  def migrate_users_and_profile_fields_to_users_settings(users_setting_record)
-    USER_FIELDS_TO_MIGRATE_TO_USERS_SETTINGS_TABLE.each do |field|
-      if USER_SETTINGS_ENUM_FIELDS.include?(field)
-        field_enums = Users::Setting.defined_enums[field]
-        users_setting_record.assign_attributes(field => field_enums[public_send(field).to_sym])
-      else
-        users_setting_record.assign_attributes(field => public_send(field))
-      end
-    end
-
-    sync_relevant_profile_fields_to_user_settings_table(users_setting_record)
-
-    users_setting_record.save
-  end
-
-  def sync_users_settings_table
-    users_setting_record = Users::Setting.create_or_find_by(user_id: id)
-
-    migrate_users_and_profile_fields_to_users_settings(users_setting_record)
-  end
-
-  def sync_users_notification_settings_table
-    users_notification_setting_record = Users::NotificationSetting.create_or_find_by(user_id: id)
-
-    USER_FIELDS_TO_MIGRATE_TO_USERS_NOTIFICATION_SETTINGS_TABLE.each do |field|
-      users_notification_setting_record.assign_attributes(field => public_send(field))
-    end
-
-    users_notification_setting_record.save
+  def create_users_settings_and_notification_settings_records
+    Users::Setting.create(user_id: id)
+    Users::NotificationSetting.create(user_id: id)
```

Comment:
you can use `create_setting` and `create_notification_setting`, they are special methods [added to Rails by `has_one`](https://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_one)

Reply:
Noice!

Reply:
TIL!